.github/workflows/lint_and_format_check.yml

```diff
@@ -0,0 +1,31 @@
+name: Lint and Format Check (Ubuntu Latest)
+
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches:
+      - "*"
+
+permissions:
+  contents: read
+
+jobs:
+  ubuntu-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3 
+      - name: Install ICU
+        run: sudo apt-get install -y libicu-dev pkg-config
+
+      - name: Install clang-format
+        run: |
+          sudo apt update && sudo apt install clang-format-15 -y
```

Comment:
We are using `clang-format-15` but `clang++-14` as `CXX` is this expected? cc @lemire

Reply:
clang-format is a standalone tool, so I don't think it needs to be tied to the actual version of clang++.
But sure there are differences in the way these major versions (14, 15) format the code and also in the "language features"/things they are able to format.